elephant/parallel/__init__.py

```diff
@@ -0,0 +1,45 @@
+"""
+The ``elephant.parallel`` package provides classes to parallelize calls to any
+user-specified function.
+
+The typical use case is calling a function many times with different
+parameters.
+
+.. note::  This parallelization module is independent from Elephant and can be
+           used to parallelize mutually independent calls to arbitrary
+           functions.
+
+Tutorial
+--------
+
+:doc:`View tutorial <../tutorials/parallel>`
+
+Run tutorial interactively:
+
+.. image:: https://mybinder.org/badge.svg
+   :target: https://mybinder.org/v2/gh/NeuralEnsemble/elephant/master?filepath=doc/tutorials/parallel.ipynb
+
+
+Available Executors
+-------------------
+
+.. autosummary::
+    :toctree: toctree/parallel/
+
+    ProcessPoolExecutor
+    MPIPoolExecutor
+    MPICommExecutor
+
+
+:copyright: Copyright 2014-2020 by the Elephant team, see `doc/authors.rst`.
+:license: Modified BSD, see LICENSE.txt for details.
+"""
+
+from .parallel import ProcessPoolExecutor, SingleProcess
+
+try:
+    from .mpi import MPIPoolExecutor, MPICommExecutor
+except ImportError:
+    # mpi4py is missing
+    # please run command `pip install -r requirements-extras.txt`
```

Comment:
I agree it's not clear.

1. If I print a message, then all other users will see the message "Cannot import blah-blah. Please install extras." while `import elephant`.

The only solution is 

2. still put there a print but remove `from . import parallel` from `__init__.py`. This will, however, disturb slightly our ecosystem - we've been always importing all modules in `__init__.py` and now we make an exception.

Anyway, I'd rather do 2.
Yes, nor?